src/nvcgo/internal/cgroup/ebpf.go

```diff
@@ -136,10 +136,10 @@ func (p *program) appendDevice(dev specs.LinuxDeviceCgroup, labelPrefix string)
 	}
 	if hasAccess {
 		p.insts = append(p.insts,
-			// if (R3 & bpfAccess == 0 /* use R2 as a temp var */) goto next
+			// if (R3 & bpfAccess != R3 /* use R2 as a temp var */) goto next
 			asm.Mov.Reg32(asm.R2, asm.R3),
```

Comment:
Both runc and crun use `R1` as a temporary variable here. Could it cause issues that we're using `R2` since we're also using `R2` in the `hasAccess` block above?

cc @klueska

Reply:
ah yes you're right ! Since we erase R2 the bpfType check in the next block will break. Interesting that the issue was never encountered before. Fixing: I'll just use a free register, R6

Reply:
done, thanks :)

Reply:
Let's rather use R1 since to align with the other implementations.

Reply:
done